src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -1820,16 +1837,29 @@ _mongoc_client_retryable_read_command_with_stream (
       /* each read command may be retried at most once */
       is_retryable = false;
 
-      if (retry_server_stream) {
-         mongoc_server_stream_cleanup (retry_server_stream);
-      }
+      {
+         mongoc_deprioritized_servers_t *const ds =
+            mongoc_deprioritized_servers_new ();
+
+         if (retry_server_stream) {
+            mongoc_deprioritized_servers_add_if_sharded (
+               ds, retry_server_stream->topology_type, retry_server_stream->sd);
+            mongoc_server_stream_cleanup (retry_server_stream);
+         } else {
+            mongoc_deprioritized_servers_add_if_sharded (
+               ds, server_stream->topology_type, server_stream->sd);
+         }
```

Comment:
Though the prior code checks for `retry_server_stream`, I expect `retry_server_stream` is always NULL since the retry only occurs once. Suggested simplification:

```c
        BSON_ASSERT (!retry_server_stream); // Retry only happens once.
        mongoc_deprioritized_servers_add_if_sharded (
           ds, server_stream->topology_type, server_stream->sd);
```

Reply:
Confirmed the suggestion works as described.